runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/TestDataflowRunner.java

```diff
@@ -102,8 +101,6 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
         job.getJobId(),
         expectedNumberOfAssertions);
 
-    assertThat(job, testPipelineOptions.getOnCreateMatcher());
```

Comment:
This makes sure test fail-fast if pipeline creation failed. Do we have alternative solutions?

Reply:
Does it do that? My understanding was that this would be a no-op by default, since OnCreateMatcher is an instance of AlwaysPassMatcher. It will only fail fast if the user sets an OnCreateMatcher, which I don't think is happening anywhere outside of TestDataflowRunnerTest.

Reply:
No, it wasn't used in any of the integration tests. I think it's okay to remove since the following `waitForxxxJobTermination` is monitoring the job status.